#!/usr/bin/env python
PACKAGE = "rsa17"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("weight_partial", double_t, 0, "Partial region weighting", 2.0, 0, 1000)
gen.add("weight_expanded", double_t, 0, "Expanded region weighting", 1.0, 0, 1000)
gen.add("weight_skeleton", double_t, 0, "Expanded region weighting", 1.0, 0, 1000)

gen.add("astar_voronoi_restricted", double_t, 0, "Voronoi Constraint: Restricted", 0.2, 0, 2)
gen.add("astar_voronoi_partial", double_t, 0, "Voronoi Constraint: Partial", 0.2, 0, 2)
gen.add("astar_voronoi_expand", double_t, 0, "Voronoi Constraint: Expand", 0.5, 0, 2)
gen.add("astar_voronoi_orphanThreshold", double_t, 0, "Voronoi Constraint: Orphan Threshold", 20, 0, 100)

# This file name must match the same name as the 3rd parameter, or catkin will keep rebuilding the header file
exit(gen.generate(PACKAGE, "crosbot_astar_reconfig", "CrosbotAStarReconfig"))